### PR TITLE
Add support for IssueProperty resource

### PR DESCRIPTION
**Pull request by turran**
Add support for adding, removing and getting issue properties as described in https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-issue-properties/#api-group-issue-properties

### CODE REVIEW

File: jira/client.py [hunk 1]
```diff
@@ -2396,6 +2397,49 @@ def add_worklog(
...
+    def issue_properties(self, issue: str) -> List[IssueProperty]:
+        """Get a list of issue property Resource from the server for an issue.
+
+        Args:
+            issue (str): ID or key of the issue to get properties from
+
+        Returns:
+            List[IssueProperty]
+        """
+        r_json = self._get_json("issue/" + str(issue) + "/properties")
```
Comment (adehad) [inline jira/client.py:2412]:
```suggestion
        r_json = self._get_json(f"issue/{issue}/properties")
```

We prefer f-strings

File: jira/client.py [hunk 2]
```diff
@@ -2396,6 +2397,49 @@ def add_worklog(
...
+
+        Args:
+            issue (str): ID or key of the issue to set the property to
+            key (str): Key of the property to set
+            data: The data to set for the property
+        Returns:
+            Response
+        """
+
+        url = self._get_url("issue/" + issue + "/properties/" + key)
```
Comment (adehad) [inline jira/client.py:2440]:
```suggestion
        url = self._get_url(f"issue/{issue}/properties/{key}")
```

(f-strings)

File: jira/resources.py [hunk 3]
```diff
@@ -290,6 +291,20 @@ def find(
...
             path = self._resource.format(id)
         url = self._get_url(path)
+        self._find_by_url(url, params)
+
+    def _find_by_url(
+        self,
+        url: str,
+        params: Optional[Dict[str, str]] = None,
+    ):
+        """Finds a resource on the specified url.
```
Comment (adehad) [inline jira/resources.py:301]: Could you elaborate in this docstring, as there is no return value, but it instead loads the url.
Perhaps the name itself can be changed.

I'm also not sure what `.find()`

File: jira/resources.py [hunk 4]
```diff
@@ -290,6 +291,20 @@ def find(
...
+
+        Args:
+            url (str): url
+            params (Optional[Dict[str, str]]): params
+
```
Comment (adehad) [inline jira/resources.py:306]:
```suggestion
```

**Review by adehad — CHANGES_REQUESTED** [review]
Excellent stuff here, few minor changes

File: jira/client.py (hunk 2 above)
Comment (turran) [reply jira/client.py:2440]: Understood. Wasn't sure which one to use given that the code had both ways. Fixing it.

File: jira/resources.py (hunk 3 above)
Comment (turran) [reply jira/resources.py:301]: Sure. The description is just a copy of the original _find(), I had to split it on two: _find() -> _find_by_url() in order to just overload one in the IssueProperty resource class. The motivation is that, despite other resources, IssueProperty JSON description does not have a "self" keyword and that has to be set by the passed in URL of the request.

File: jira/client.py (hunk 2 above)
Comment (adehad) [reply jira/client.py]: ahh, we tried to cleanse the code base using flynt or something, but maybe we didn't do a good enough job there, sorry

File: jira/resources.py (hunk 3 above)
Comment (adehad) [reply jira/resources.py]: Cool, that is interesting. 

I'm a bit cautious of the additional request that needs to be made for this edge case. But tests are happy so probably will be alright, I'll give it a think but should be good

**Review by adehad — APPROVED** [review]

**Comment by adehad** [comment]
will be trying to get some bugfixes in for a patch release, so will aim to merge this after that

**Comment by adehad** [comment]
Thanks for the (very high quality) PR ! Now included in `3.4.0`

Sorry the original one got buried with the CI issues, we should be in a better place now for quicker reviews and deployments!
